app-testsuite/src/run.ts

```diff
@@ -240,26 +240,31 @@ class App {
   }
 }
 
+function createApp(testDir: string): Promise<App> {
+  return withRegistry(testDir, async () => {
+    const app = new App(testDir, 'my-app');
+    await app.create();
+    return app;
+  });
+}
+
 async function main() {
   const testDir = await mkdtemp(path.resolve(tmpdir(), 'test'), { encoding: 'utf8' });
   log('Created test dir', testDir);
   try {
-    await withRegistry(testDir, async () => {
-      const app = new App(testDir, 'my-app');
-      await app.create();
-      const runMode = process.env.APP_E2E_RUN_MODE || 'local';
-      const run = runMode === 'remote' ? app.runRemote : app.runLocal;
-      await run.bind(app)(async (baseUrl) => {
-        log('Running tests');
-        await spawn('npx', ['cypress', 'run'], {
-          cwd: path.resolve(__dirname, '..'),
-          stdio: 'inherit',
-          shell,
-          env: {
-            ...process.env,
-            CYPRESS_baseUrl: baseUrl,
-          },
-        });
+    const app = await createApp(testDir);
+    const runMode = process.env.APP_E2E_RUN_MODE || 'local';
+    const run = runMode === 'remote' ? app.runRemote : app.runLocal;
+    await run.bind(app)(async (baseUrl) => {
```

Comment:
`runMode` can be "bound" into the app as a property.

Reply:
Changed a bit